File: test/counter/counter_test.js
const specimen = require("../../js/dist");

let counter = 0;

specimen.run(
  (s, tile) => {
    expected = tile.expected_count;
    if (expected !== undefined) {
      s.expectEqual(counter, Number(expected), "count comparison");
    }
    counter += 1;
  },
  [specimen.readLocalFile("counter_data.yaml", __dirname)]
);

File: test/counter/counter_test.go
package zoo_test

import (
	"strconv"
	"testing"

	specimen "github.com/ditrit/specimen/go"
)

var counter = 0

func TestCounting(t *testing.T) {
	specimen.Run(
		t,
		func(s *specimen.S, input specimen.Dict) {
			expectedString, found := input["expected_count"]
			if found {
				expected, err := strconv.Atoi(expectedString)
				if err != nil {
					panic(err)
				}

				s.ExpectEqual(counter, expected, "count comparison")
			}
			counter += 1
		},
		[]specimen.File{
			specimen.ReadLocalFile("counter_data.yaml"),
		},
	)
}
